py-polars/tests/unit/test_queries.py

```diff
@@ -370,3 +372,115 @@ def test_shift_drop_nulls_10875() -> None:
     assert pl.LazyFrame({"a": [1, 2, 3]}).shift(1).drop_nulls().collect()[
         "a"
     ].to_list() == [1, 2]
+
+
+def test_utf8_date() -> None:
+    df = pl.DataFrame({"x1": ["2021-01-01"]}).with_columns(
+        **{"x1-date": pl.col("x1").cast(pl.Date)}
+    )
+    expected = pl.DataFrame({"x1-date": [date(2021, 1, 1)]})
+    out = df.select(pl.col("x1-date"))
+    assert out.shape == (1, 1)
+    assert out.dtypes == [pl.Date]
```

Comment:
Looks like you're already using the assert frame equal function - if so then you don't need these assertions on length and dtypes